util/execdetails/execdetails.go

```diff
@@ -410,6 +421,23 @@ func (e *RuntimeStats) SetRowNum(rowNum int64) {
 	atomic.StoreInt64(&e.rows, rowNum)
 }
 
+// SetConcurrencyInfo sets the concurrency information.
+func (e *RuntimeStats) SetConcurrencyInfo(name string, num int) {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+	e.concurrency = append(e.concurrency, concurrencyInfo{concurrencyName: name, concurrencyNum: num})
+}
+
 func (e *RuntimeStats) String() string {
-	return fmt.Sprintf("time:%v, loops:%d, rows:%d", time.Duration(e.consume), e.loop, e.rows)
+	result := fmt.Sprintf("time:%v, loops:%d, rows:%d", time.Duration(e.consume), e.loop, e.rows)
+	if len(e.concurrency) > 0 {
+		for _, concurrency := range e.concurrency {
+			if concurrency.concurrencyNum > 0 {
+				result += fmt.Sprintf(", %s:%d", concurrency.concurrencyName, concurrency.concurrencyNum)
+			} else {
+				result += fmt.Sprintf(", %s:OFF", concurrency.concurrencyName)
+			}
+		}
```

Comment:
We should lock this section.

Reply:
Why we should lock this? Does anywhere cause the data race?

Reply:
My fault, it's acceptable not locking this section. Should we add a comment for this method and state that it is not thread safe?